web/src/poll_widget.ts

```diff
@@ -1,20 +1,50 @@
 import $ from "jquery";
 
 import {PollData} from "../shared/src/poll_data";
+import type {
+    InboundData,
+    NewOptionOutboundData,
+    QuestionOutboundData,
+    VoteOutboundData,
+} from "../shared/src/poll_data";
 import render_widgets_poll_widget from "../templates/widgets/poll_widget.hbs";
 import render_widgets_poll_widget_results from "../templates/widgets/poll_widget_results.hbs";
 
 import * as blueslip from "./blueslip";
 import {$t} from "./i18n";
 import * as keydown_util from "./keydown_util";
 import * as people from "./people";
+import type {Message} from "./types";
+
+type Event = {sender_id: number; data: InboundData};
+
+type ExtraData =
+    | {
+          question?: string | undefined;
+          options?: never[] | undefined;
+      }
+    | undefined;
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+    interface JQuery {
+        handle_events(events: Event[]): void;
+    }
+}
 
 export function activate({
     $elem,
     callback,
-    extra_data: {question = "", options = []} = {},
+    extra_data: {question = "", options = []} = {}, // ?
     message,
-}) {
+}: {
+    $elem: JQuery;
+    callback: (
+        data: NewOptionOutboundData | QuestionOutboundData | VoteOutboundData | undefined,
+    ) => void; // todo check whether can take undefined
```

Comment:
Yeah, I think accepting `undefined` here is valid since for `question.outbound` function in `poll_data` we indeed have a code path that returns undefined and not an event.
https://github.com/zulip/zulip/blob/31b58792422bbd07286897fbfa347e29f9647d0d/web/shared/src/poll_data.ts#L164-L174